pylearn2/models/mlp.py

```diff
@@ -3059,6 +3079,14 @@ def set_input_space(self, space):
 
         rng = self.mlp.rng
 
+        if self.pad != (0, 0):
+            output_shape = \
+                [int(np.ceil((i_sh + 2. * k_pad - k_sh) / float(k_st))) + 1
```

Comment:
Actually, I'm not sure the `ceil` part is right, as it does not give the same result as the integer division below when pad is `(0, 0)`.